diff -Nupr -X linux-2.6.33-lpc313x/Documentation/dontdiff -x modules.builtin -x hash -x '*.gzip' linux-2.6.33-lpc313x/arch/arm/mach-lpc313x/ea313x.c gnublin-linux-2.6.33/arch/arm/mach-lpc313x/ea313x.c
--- linux-2.6.33-lpc313x/arch/arm/mach-lpc313x/ea313x.c	2012-01-04 10:35:17.000000000 +0100
+++ gnublin-linux-2.6.33/arch/arm/mach-lpc313x/ea313x.c	2012-01-04 11:57:40.000000000 +0100
@@ -51,7 +51,7 @@ static struct lpc313x_mci_irq_data irq_d
 
 static int mci_get_cd(u32 slot_id)
 {
-	return gpio_get_value(GPIO_MI2STX_BCK0);
+	return gpio_get_value(GPIO_MNAND_RYBN2);
 }
 
 static irqreturn_t ea313x_mci_detect_interrupt(int irq, void *data)
@@ -74,8 +74,9 @@ static int mci_init(u32 slot_id, irq_han
 
 	/* enable power to the slot */
 	gpio_set_value(GPIO_MI2STX_DATA0, 0);
+
 	/* set cd pins as GPIO pins */
-	gpio_direction_input(GPIO_MI2STX_BCK0);
+	gpio_direction_input(GPIO_MNAND_RYBN2);
 
 	/* select the opposite level senstivity */
 	level = mci_get_cd(0)?IRQ_TYPE_LEVEL_LOW:IRQ_TYPE_LEVEL_HIGH;
diff -Nupr -X linux-2.6.33-lpc313x/Documentation/dontdiff -x modules.builtin -x hash -x '*.gzip' linux-2.6.33-lpc313x/arch/arm/mach-lpc313x/include/mach/irqs.h gnublin-linux-2.6.33/arch/arm/mach-lpc313x/include/mach/irqs.h
--- linux-2.6.33-lpc313x/arch/arm/mach-lpc313x/include/mach/irqs.h	2012-01-04 10:35:17.000000000 +0100
+++ gnublin-linux-2.6.33/arch/arm/mach-lpc313x/include/mach/irqs.h	2012-01-02 23:30:14.000000000 +0100
@@ -143,9 +143,12 @@
 #define BOARD_IRQ_EVENT_MAP	{ \
 	CHIP_IRQ_EVENT_MAP \
 	{IRQ_DM9000_ETH_INT, EVT_mNAND_RYBN3, EVT_ACTIVE_HIGH}, \
-	{IRQ_SDMMC_CD, EVT_mI2STX_BCK0, EVT_ACTIVE_LOW}, \
+  {IRQ_SDMMC_CD, EVT_mNAND_RYBN2, EVT_ACTIVE_LOW}, \
 	{IRQ_EA_VBUS_OVRC, EVT_I2SRX_WS0, EVT_ACTIVE_LOW}, \
 	}
+  //{IRQ_SDMMC_CD, EVT_mI2STX_BCK0, EVT_ACTIVE_LOW}, --hh
+
+
 /* Following defines group the board IRQs into 4 IRQ_EVNTR groups.
    IRQ_EVT_ROUTERx IRQ is generated when event in the corresponding 
    group triggers.
